### PR TITLE
Public lobbies

### PR DESCRIPTION
**Pull request by VictorSchneuwly**
Add the possibility to join public lobbies.

### CODE REVIEW

File: app/src/main/java/com/github/fribourgsdp/radio/game/prep/JoinGameActivity.kt [hunk 1]
```diff
@@ -106,4 +115,36 @@ open class JoinGameActivity : MyAppCompatActivity() {
             joinErrorView.visibility = View.VISIBLE
         }
     }
+
+    private fun initPublicLobbiesDisplay() {
+        // init the cards
+        lobbiesRecyclerView.layoutManager = LinearLayoutManager(this)
+        val adapter = PublicLobbiesAdapter(this, this.db).apply {
```
Comment (nathanduchesne) [inline app/src/main/java/com/github/fribourgsdp/radio/game/prep/JoinGameActivity.kt:122]: Here could you not replace this.db with db?

File: app/src/main/java/com/github/fribourgsdp/radio/game/prep/JoinGameActivity.kt [hunk 2]
```diff
@@ -42,6 +46,8 @@ open class JoinGameActivity : MyAppCompatActivity() {
         joinButton = findViewById(R.id.joinGameButton)
         joinWithQRCodeButton = findViewById(R.id.joinWithQRCode)
         joinErrorView = findViewById(R.id.joinErrorView)
+        lobbiesRecyclerView = findViewById(R.id.publicLobbiesRecyclerView)
+        spinner = findViewById(R.id.lobbySortSpinner)
```
Comment (nathanduchesne) [inline app/src/main/java/com/github/fribourgsdp/radio/game/prep/JoinGameActivity.kt:50]: If you have time you could do an initViews() as we have regularly done in other files, as well as an initBehaviours() or initListeners() for setting the listeners.

Comment (VictorSchneuwly) [reply app/src/main/java/com/github/fribourgsdp/radio/game/prep/JoinGameActivity.kt:50]: Good idea 👍🏻

Comment (VictorSchneuwly) [reply app/src/main/java/com/github/fribourgsdp/radio/game/prep/JoinGameActivity.kt:50]: Done in pr #242

File: app/src/main/java/com/github/fribourgsdp/radio/game/prep/JoinGameActivity.kt (hunk 1 above)
Comment (VictorSchneuwly) [reply app/src/main/java/com/github/fribourgsdp/radio/game/prep/JoinGameActivity.kt:122]: Done in pr #242

**Comment by VictorSchneuwly** [comment]
Some bugs were corrected in the new pr #242

**Pull request closed without merging**